### PR TITLE
Documented remaining Attributes of classes and functions

### PR DESCRIPTION
**Pull request by v-parmar**
#### Reference Issue
#716

#### What does this PR implement/fix? Explain your changes.
I have added the missing documentation for attributes of classes and functions.

### CODE REVIEW

File: openml/tasks/functions.py
Comment (LennartPurucker) [inline openml/tasks/functions.py:233]:
```suggestion
    """Returns a dictionary or a Pandas DataFrame with information about OpenML tasks.
```

File: openml/tasks/split.py
Comment (LennartPurucker) [inline openml/tasks/split.py:139]:
```suggestion
        """Generates a new OpenML dataset object from input data and cross-validation settings.
```

File: openml/tasks/split.py
Comment (LennartPurucker) [inline openml/tasks/split.py:160]:
```suggestion
        """Returns the specified data split from the CrossValidationSplit object.
```

**Review by LennartPurucker — COMMENTED** [review]
I just added some minor changes to keep things consistent. Otherwise, all looks good to me, thank you!

**Comment by v-parmar** [comment]
Thank you for the review. I have updated the changes you mentioned.

**Review by LennartPurucker — APPROVED** [review]

**Comment by LennartPurucker** [comment]
Thanks! Before we can merge this, we need to fix the pre-commit (formatting). 
You will have to run this locally or let the bot fix the formatting. 

However, it seems that the bot cannot fix it, could it be that the PR's "Allow edits by maintainers" box was unchecked?

**Comment by v-parmar** [comment]
I have checked the "Allow edits by maintainers" box.

**Comment by LennartPurucker** [comment]
Ah yes, the pre-commit fails due to an indent error. The tests are failing for the same reason. 

The specific error message is: `error: cannot format openml/tasks/task.py: unindent does not match any outer indentation level (<tokenize>, line 584)`. 

Could you try to fix this? Once you have fixed this, I think the workflows should run through and the reformatter too. 

If you can and want to make sure, you could try to run the [pre-commit locally](https://github.com/openml/openml-python/blob/develop/CONTRIBUTING.md#pull-request-checklist) before committing, but this might not be needed if it is only this one bug.

**Comment by v-parmar** [comment]
Hi @LennartPurucker, I think the issue is resolved and PR is able to merge.

**Comment by LennartPurucker** [comment]
Hi @v-parmar, the pre-commit is still [failing ](https://results.pre-commit.ci/run/github/17939040/1692594777.jyY7kXeZSYSLOxv3b3V3bQ) and I cannot merge it until these errors are resolved. 

These errors mostly seem to be that lines are too long and the pre-commit bot does not fix such errors. You will have to run the pre-commit locally or fix these manually. I have found no way to do this from my side as part of the PR. If I find the time I can run the pre-commit and commit it to your branch if that works but I am unsure about that right now.

**Comment by v-parmar** [comment]
Hi @LennartPurucker, I have resolved that lines are too long error.

**Comment by LennartPurucker** [comment]
Hi @v-parmar, sorry for taking so long to respond. 

I have checked your changes and you have taken care of the problems previously mentioned. 

But the pre-commit is still failing due to incorrect type annotations. 
I will try to fix these by the end of next week and then merge your changes.

Thank you once again for all the effort you put into this!

**Review by LennartPurucker — APPROVED** [review]

**Comment by LennartPurucker** [comment]
Heyho, I finally came around to fix the mypy errors in the pre-commit. I am waiting for the tests to finish and then merge the PR.

**Comment by LennartPurucker** [comment]
@v-parmar, I have not found a method to push to your branch. Hence, could you add ` * ADD #716: add documentation for remaining attributes of classes and functions.` to `doc/progress.rst` below ` * MAINT #1280: Use the server-provided ``parquet_url`` instead of ``minio_url`` to determine the location of the parquet file.`. 

@PGijsbers, the tests seem to be failing due to server problems or am I missing something? 
If so, we could still merge this IMO as the PR does not change code except for type hints.

**Review by PGijsbers — APPROVED** [review]
Yes, this seems fine to me. The test errors seem to be due to test server state. I believe it was recently restarted which explains some of the failures I see. I think there's also a small chance that a wrong version of the test server data was deployed (based on the unknown task type). Either way, it does not look like the failures are related to the changes in the PR.

**Comment by v-parmar** [comment]
Is this PR ready to merge or are there any further changes needed?

**Comment by LennartPurucker** [comment]
@v-parmar, this completes the PR, thank you for your work and patience!

**Pull request merged**
